prime-dotnet-webapi/Services/OrganizationService.cs

```diff
@@ -118,6 +140,52 @@ public async Task<int> CreateOrganizationAsync(int partyId)
             return organization.Id;
         }
 
+        public async Task<Organization> ClaimOrganizationAsync(OrganizationClaimViewModel claimOrganization)
+        {
+            var organization = await GetOrganizationByPecAsync(claimOrganization.PEC);
+            if (organization == null)
+            {
+                return null;
+            }
+
+            var organizationCLaim = new OrganizationClaim
+            {
+                OrganizationId = organization.Id,
+                NewSigningAuthorityId = claimOrganization.PartyId,
+                ProvidedSiteId = claimOrganization.PEC,
+                Details = claimOrganization.ClaimDetail
+            };
+
+            _context.OrganizationClaims.Add(organizationCLaim);
+            await _context.SaveChangesAsync();
+
+            await _businessEventService.CreateOrganizationEventAsync(organization.Id, claimOrganization.PartyId, "Organization Claim Created");
+
+            return organization;
+        }
+
+        public async Task<bool> OrganizationClaimExistsAsync(OrganizationClaimSearchOptions searchOptions)
+        {
+            // return false if searchOptions is invalid
+            if (searchOptions == null || string.IsNullOrEmpty(searchOptions.Pec) && string.IsNullOrEmpty(searchOptions.UserId))
+            {
+                return false;
+            }
+
+            var userId = Guid.Empty;
+            Guid.TryParse(searchOptions.UserId, out userId);
```

Comment:
Rather than accepting a string and then trying to parse it into a `Guid`, you can make the type on the View Model a Guid